components/button.css

```diff
@@ -0,0 +1,187 @@
+.btn {
+  background-color: transparent;
+  border: none;
+  cursor: pointer;
+  outline: none;
+  white-space: nowrap;
+}
+
+.btn:focus {
+  outline: 2px double var(--bg-primary);
+  outline-offset: 2px;
+}
+
+.btn i {
+  margin: 0 0.4rem;
+}
+
+/* buttons */
+.btn:not(.btn-icon) {
+  border: 1px solid transparent;
+  border-radius: 0.475rem;
+  padding: 0.7rem 0.8rem;
+  color: var(--white-text);
+  transition: background-color 0.2s linear;
+}
+
+.btn[disabled] {
+  opacity: 0.5;
+  pointer-events: none;
+}
+
+.btn-primary {
+  background-color: var(--bg-primary);
+}
+
+.btn-primary:hover {
+  background-color: #013f9b;
+}
+
+.btn.btn-primary-outline {
+  border: 1px solid var(--bg-primary);
+  color: black;
+}
+
+.btn-primary-outline:hover {
+  background-color: var(--bg-primary);
+  color: var(--white-text);
+}
+
+.btn.btn-secondary {
+  background-color: var(--bg-secondary);
+  color: black;
+}
+
+.btn-secondary:hover {
+  background-color: #6a8686;
+}
+
+.btn.btn-secondary-outline {
+  border: 1px solid var(--bg-secondary);
+  color: black;
+}
+
+.btn.btn-secondary-outline:hover {
+  background-color: var(--bg-secondary);
+}
+
+.btn-success {
+  background-color: var(--btn-success);
+}
+
+.btn-success:hover {
+  background-color: rgb(17, 88, 58);
+}
+
+.btn-danger {
+  background-color: var(--btn-danger);
+}
+
+.btn-danger:hover {
+  background-color: #ac0011;
+}
+
+.btn.btn-warning {
+  background-color: var(--btn-warning);
+  color: black;
+}
+
+.btn.btn-warning:hover {
+  background-color: #f0c33e;
+}
+
+.btn.btn-info {
+  background-color: var(--btn-info);
+  color: black;
+}
+
+.btn-info:hover {
+  background-color: #12bbdd;
+}
+
+.btn.btn-lg {
+  padding: 1.2rem 2rem !important;
+}
+
+.btn.btn-sm {
+  padding: 0.5rem 0.6rem !important;
+}
+
+.btn.btn-link {
+  text-decoration: underline;
+  text-underline-offset: 2px;
+  text-decoration-color: var(--bg-primary);
+  color: initial;
+  font-size: 1.2rem;
+}
+
+.btn.btn-link:hover {
+  color: var(--bg-primary);
+}
+
+.btn.icon-btn-primary i {
+  color: var(--bg-primary);
+  font-size: 1.2rem;
+}
+
+/* toggle btn */
+
+.toggle-btn {
+  box-sizing: border-box;
+  display: inline-block;
+  height: 40px;
+  width: 90px;
```

Comment:
Try to maintain units consistency.

Reply:
The toggle button doesn't need to increase with the root font-size hence the values are added in pixels.

Reply:
Okay!! Looks nice it's ready to merge.